src/WPF/ArcGISRuntime.WPF.Samples/Samples/Layers/ExportTiles/ExportTiles.xaml.cs

```diff
@@ -0,0 +1,264 @@
+﻿// Copyright 2017 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using Esri.ArcGISRuntime.Geometry;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Symbology;
+using Esri.ArcGISRuntime.Tasks.Offline;
+using Esri.ArcGISRuntime.UI;
+using System;
+using System.Linq;
+using System.Windows;
+using System.Windows.Media;
+
+namespace ArcGISRuntime.WPF.Samples.ExportTiles
+{
+    public partial class ExportTiles
+    {
+        // URL to the service tiles will be exported from
+        private Uri _serviceUri = new Uri("https://sampleserver6.arcgisonline.com/arcgis/rest/services/World_Street_Map/MapServer");
+
+        public ExportTiles()
+        {
+            InitializeComponent();
+
+            // Call a function to set up the map
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            // Create the tile layer
+            ArcGISTiledLayer layer = new ArcGISTiledLayer(_serviceUri);
+
+            // Load the layer
+            await layer.LoadAsync();
+
+            // Create the basemap with the layer
+            Map myMap = new Map(new Basemap(layer));
+
+            // Assign the map to the mapview
+            MyMapView.Map = myMap;
+
+            // Create a new symbol for the extent graphic
+            SimpleLineSymbol lineSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle.Solid, Colors.Red, 2);
+
+            // Create graphics overlay for the extent graphic and apply a renderer
+            GraphicsOverlay extentOverlay = new GraphicsOverlay();
+            extentOverlay.Renderer = new SimpleRenderer(lineSymbol);
+
+            // Add graphics overlay to the map view
+            MyMapView.GraphicsOverlays.Add(extentOverlay);
+
+            // Subscribe to changes in the mapview's viewpoint so the preview box can be kept in position
+            MyMapView.ViewpointChanged += MyMapView_ViewpointChanged;
+        }
+
+        private void MyMapView_ViewpointChanged(object sender, EventArgs e)
+        {
+            UpdateMapExtent();
+        }
+
+        /// <summary>
+        /// Function used to keep the overlaid preview area marker in position
```

Comment:
Perhaps some more comments here as to what this function does. Something like:
Each time the user changes the extent of the MapView (either by a zoom in/out or a pan) this function fires. When the extent changes, the existing red box (aka. envelope) that is overlaid on the MapView gets updated to be 80% of the current extent. It is the extent of the red box that appears on the MapView that is used to define the extent of tiles that are retrieved from the ExportTileCacheJob.